pkg/controlplane/instance.go

```diff
@@ -631,17 +630,6 @@ func (c completedConfig) newKubernetesServiceControllerConfig(client kubernetes.
 		return nil, fmt.Errorf("failed to get listener address: %w", err)
 	}
 
-	// The "kubernetes.default" Service is SingleStack based on the configured ServiceIPRange.
-	// If the bootstrap controller reconcile the kubernetes.default Service and Endpoints, it must
-	// guarantee that the Service ClusterIP and the associated Endpoints have the same IP family, or
-	// it will not work for clients because of the IP family mismatch.
-	// TODO: revisit for dual-stack https://github.com/kubernetes/enhancements/issues/2438
-	if c.ExtraConfig.EndpointReconcilerType != reconcilers.NoneEndpointReconcilerType {
-		if netutils.IsIPv4CIDR(&c.ExtraConfig.ServiceIPRange) != netutils.IsIPv4(c.GenericConfig.PublicAddress) {
```

Comment:
we need to be sure this gets the IP after it is obtained, since it may be obtained from the default route of the node if not set 

https://github.com/kubernetes/kubernetes/blob/8d0e46a5ec38faed3e7277710103b37fa9140378/staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go#L126

Reply:
not sure I understand. What's missing?

Reply:
answered here https://github.com/kubernetes/kubernetes/pull/119081#discussion_r1252364970